investigations/LOKI/loki_performance.md

```diff
@@ -0,0 +1,122 @@
+# LOKI Live Reduction Performance in Mantid
+
+## Motivation
+
+The LOKI instrument for small angle neutron scattering is scheduled to be one of the first instruments to come online for the ESS. As such, it is important that the day 1 requirements for the software are satisfied before the instrument enters the commissioning phase. LOKI will be ~1Mpixels with an incident flux between 10<sup>5</sup>Hz and 10<sup>8</sup>Hz (worst case). This represents a significant processing and storage challenge for the ESS. Since Mantid has been chosen as the data-reduction platform at the ESS, it is critical that the software can cope with these data rates. This document outlines the current performance of the Mantid framework and recommendations for improvements which may allow performance to meet the requirements.
+
+## Scope
+This document only covers performance benchmarks for the LOKI instrument based on the geometry described in the proceeding section. A general benchmarking exercise of the live data reduction in Mantid was performed [here](https://github.com/DMSC-Instrument-Data/documents/blob/master/investigations/Live%20Reduction/LiveReductionInvestigation.md) and this document may be referenced within.
+
+## LOKI Geometry
+The LOKI Geometry used for these tests are as specified in the engineering diagrams below. The instrument consists of 864 tubes (6048 straws) with an intrinsic resolution of 5mm. Straw lengths vary between 1.2m and 0.5m which gives a total pixel count of 1,245,880pixels.
+![geometry](loki_geometry.png)
+![mantid](loki_mantid.png)
+
+## Build-out Phases
+
+Currently "Day 1" operations for LOKI are set to commence November 2022. The initial instrument will contain the rear detector bank and 2 panels (one horizontal and one vertical) in the front and middle banks each. The resulting instrument will consist of ~473K pixels. However based on initial simulations, most of the detected neutron beam will be concentrated on the rear bank. The remaining banks are currently set to be added to the final instrument between 1-5 years beyond 2022. Hot commissioning and early science for this instrument are set for mid 2023 with full operations at the start of 2025.
+
+## Hardware 
+
+The hardware used for these tests is as follows:
+
+- 128GB RAM
+- Intel(R) Core(TM) i9-9920X CPU @ 3.50GHz
```

Comment:
Don't think so.

Reply:
Might be worth mentioning OpenMP support must be present. We don't have this on Mac, not sure about Windows.

However, since the ESS are set on using CentOS it may not be important enough to mention.